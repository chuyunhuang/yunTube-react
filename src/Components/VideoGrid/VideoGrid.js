import React from 'react';
import './VideoGrid.scss';
import { VideoPreview } from '../VideoPreview/VideoPreview';
import { Divider } from 'semantic-ui-react';

export function VideoGrid(props) {
  return (
    <React.Fragment>
      <h4>Trending</h4>
      <div className="video-grid">
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
        <VideoPreview />
      </div>
      {Divider}
    </React.Fragment>
  )
}
