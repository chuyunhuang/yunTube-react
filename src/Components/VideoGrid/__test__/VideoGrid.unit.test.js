import React from 'react';
import { shallow } from 'enzyme';
import { VideoGrid } from '../VideoGrid';

describe('VideoGrid', () => {
  test('renders VideoGrid without props', () => {
    const wrapper = shallow(
      <VideoGrid />
    )
    expect(wrapper).toMatchSnapshot();
  });

  test('renders VideoGrid with title prop', () => {
    const wrapper = shallow(
      <VideoGrid title="testing" />
    )
    expect(wrapper).toMatchSnapshot();
  });

  test('renders VideoGrid without divider', () => {
    const wrapper = shallow(
      <VideoGrid hideDivider={true} />
    )
    expect(wrapper).toMatchSnapshot();
  });

});