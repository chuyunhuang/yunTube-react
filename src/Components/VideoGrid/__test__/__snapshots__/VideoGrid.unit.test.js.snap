// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoGrid renders VideoGrid with title prop 1`] = `
<Fragment>
  <VideoGridHeader
    title="testing"
  />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`VideoGrid renders VideoGrid without divider 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
</Fragment>
`;

exports[`VideoGrid renders VideoGrid without props 1`] = `
<Fragment>
  <VideoGridHeader />
  <div
    className="video-grid"
  >
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
    <VideoPreview />
  </div>
  <Divider />
</Fragment>
`;
